src/Serialization/main.jl

```diff
@@ -81,36 +103,103 @@ function decodeType(input::String)
     end
 end
 
+################################################################################
+# Encoding helper functions
+
+@doc raw"""
+    is_basic_serialization_type(::Type)
 
+During the serialization of types of the form `Vector{T}`, entries
+of type `T` will either be serialized as strings if `is_basic_serialization_type`
+returns `true`, or serialized as a dict provided the serialization for such a `T`
+exists. If `Vector{T}` is serialized with `is_basic_serialization_type(T) = true`
+then the `entry_type` keyword is used to store the type `T` as a property of
+the vector.
+
+# Examples
+
+```jldoctest
+julia> is_basic_serialization_type(ZZRingElem)
+true
+```
+"""
+is_basic_serialization_type(::Type) = false
+is_basic_serialization_type(::Type{ZZRingElem}) = true
+is_basic_serialization_type(::Type{QQFieldElem}) = true
+is_basic_serialization_type(::Type{Bool}) = true
+is_basic_serialization_type(::Type{Symbol}) = true
+# this deals with int32, int64 etc.
+is_basic_serialization_type(::Type{T}) where T <: Number = isconcretetype(T)
+
+# ATTENTION
+# We need to distinguish between data with a globally defined normal form and data where such a normal form depends on some parameters.
+# In particular, this does NOT ONLY depend on the type; see, e.g., FqField.
+
+# Objects have a elements that have a basic encoding are fundamental objects. They are "atomic"
```

Comment:
"Objects have a elements" ??

Is this only talking about "things that have elements" (like rings, groups, ...) or about anything that has basic serialization?